cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1001,6 +1025,17 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onRebalancing(Reco
                                             LOGGER.errorCr(reconciliation, "Cruise Control stopping execution failed", e.getCause());
                                             p.fail(e.getCause());
                                         });
+                                } else if (rebalanceAnnotation(currentKafkaRebalance) == KafkaRebalanceAnnotation.refresh) {
+                                    LOGGER.debugCr(reconciliation, "Stopping current Cruise Control rebalance user task since spec has been updated and requesting a new one");
+                                    vertx.cancelTimer(t);
+                                    apiClient.stopExecution(host, CruiseControl.REST_API_PORT)
```

Comment:
A couple thoughts about the partitions that have already been moved by the rebalance operation which we are stopping. 
- Will the newly requested proposal account for these moved partitions
- Is there any reason we would want to roll back the changes/movements? (Not that we can do anything here as part of this PR concerning this, that functionality would need its own PR)

Make me second guess our direction and think if it would be better if instead of stopping and requesting new proposal in the `Rebalacing` state we just log and update the `KafkaRebalance` status section with an error. Just a thought, let me know what you think!

Reply:
@kyguy I will have to test this use case. But if I am not wrong if we stop the rebalance and ask for a new rebalance. I think it will create a fresh proposal based on the current situation so it should account for the moved partition in my opinion.